kde-apps/kdepim-runtime/kdepim-runtime-9999.ebuild

```diff
@@ -16,65 +16,61 @@ HOMEPAGE="https://apps.kde.org/kontact/"
 LICENSE="GPL-2+ LGPL-2.1+"
 SLOT="6"
 KEYWORDS=""
-IUSE="speech"
+IUSE="speech webengine"
 
 RESTRICT="test"
 
-# TODO kolab
+# TODO etebase, kolab
 RDEPEND="
 	>=app-crypt/qca-2.3.7:2[qt6]
 	dev-libs/cyrus-sasl:2
-	dev-libs/libical:=
 	>=dev-libs/qtkeychain-0.14.1-r1:=[qt6]
-	>=dev-qt/qtbase-${QTMIN}:6[dbus,gui,network,widgets,xml]
+	>=dev-qt/qtbase-${QTMIN}:6[dbus,network,widgets]
 	>=dev-qt/qtnetworkauth-${QTMIN}:6
-	>=dev-qt/qtwebengine-${QTMIN}:6[widgets]
 	>=kde-apps/akonadi-${PVCUT}:6
 	>=kde-apps/akonadi-calendar-${PVCUT}:6
 	>=kde-apps/akonadi-contacts-${PVCUT}:6
 	>=kde-apps/akonadi-mime-${PVCUT}:6
 	>=kde-apps/akonadi-notes-${PVCUT}:6
+	>=kde-apps/akonadi-${PVCUT}:6
 	>=kde-apps/kcalutils-${PVCUT}:6
 	>=kde-apps/kidentitymanagement-${PVCUT}:6
 	>=kde-apps/kimap-${PVCUT}:6
 	>=kde-apps/kldap-${PVCUT}:6
 	>=kde-apps/kmailtransport-${PVCUT}:6
 	>=kde-apps/kmbox-${PVCUT}:6
 	>=kde-apps/kmime-${PVCUT}:6
-	>=kde-apps/libkdepim-${PVCUT}:6
 	>=kde-apps/libkgapi-${PVCUT}:6
 	>=kde-frameworks/kcalendarcore-${KFMIN}:6
 	>=kde-frameworks/kcmutils-${KFMIN}:6
-	>=kde-frameworks/kcodecs-${KFMIN}:6
-	>=kde-frameworks/kcompletion-${KFMIN}:6
 	>=kde-frameworks/kconfig-${KFMIN}:6
 	>=kde-frameworks/kconfigwidgets-${KFMIN}:6
 	>=kde-frameworks/kcontacts-${KFMIN}:6
-	>=kde-frameworks/kcoreaddons-${KFMIN}:6
 	>=kde-frameworks/kdav-${KFMIN}:6
-	>=kde-frameworks/kholidays-${KFMIN}:6
-	>=kde-frameworks/ki18n-${KFMIN}:6
 	>=kde-frameworks/kio-${KFMIN}:6
 	>=kde-frameworks/kitemmodels-${KFMIN}:6
-	>=kde-frameworks/kjobwidgets-${KFMIN}:6
 	>=kde-frameworks/knotifications-${KFMIN}:6
 	>=kde-frameworks/knotifyconfig-${KFMIN}:6
-	>=kde-frameworks/kservice-${KFMIN}:6
 	>=kde-frameworks/ktextwidgets-${KFMIN}:6
 	>=kde-frameworks/kwallet-${KFMIN}:6
-	>=kde-frameworks/kwidgetsaddons-${KFMIN}:6
 	>=kde-frameworks/kwindowsystem-${KFMIN}:6
-	>=kde-frameworks/kxmlgui-${KFMIN}:6
 	speech? ( >=dev-qt/qtspeech-${QTMIN}:6 )
+	webengine? ( >=dev-qt/qtwebengine-${QTMIN}:6[widgets] )
 "
 DEPEND="${RDEPEND}
-	test? ( >=kde-apps/kimap-${PVCUT}:6[test] )
+	test? (
+		>=kde-apps/akonadi-${PVCUT}:6[tools]
+		>=kde-apps/kimap-${PVCUT}:6[test]
+	)
 "
 
 src_configure() {
 	local mycmakeargs=(
+		-DKDEPIM_RUN_AKONADI_TEST=$(usex test)
+		-DCMAKE_DISABLE_FIND_PACKAGE_Etebase=ON
 		-DCMAKE_DISABLE_FIND_PACKAGE_Libkolabxml=ON
 		$(cmake_use_find_package speech Qt6TextToSpeech)
+		$(cmake_use_find_package webengine Qt6WebEngineWidgets)
```

Comment:
The effort is laudable but there is no KDE PIM without dev-qt/qtwebengine, so you might as well make that unconditional.

See also:
https://invent.kde.org/pim/kdepim-runtime/-/commit/7e348558b89a02ae1a3c20565ad2eba3b91f31da

This is not really supposed to be user-visible.